src/lib.rs

```diff
@@ -516,6 +486,30 @@ impl Client {
         }
     }
 
+    async fn publish_and_wait<T: DeserializeOwned>(
```

Comment:
nit: does this function need to be instrumented for proper profiling and tracing?

Reply:
It should be included in the parent span. Tested locally by adding a debug message and confirming it shows up in the parent